pkg/storage/tsdb/caching_bucket.go

```diff
@@ -0,0 +1,89 @@
+package tsdb
```

Comment:
Should we name this file `chunks_cache.go` now?

Reply:
I plan to extend this file with metadata caching config soon (separate config struct, with separate cache config, but handled by the same instance of `CachingBucket`), so I would keep it as is for now. WDYT?

Reply:
Sure, it's OK.